zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/adhocsubprocess/parameter/FromAiFeelFunctionCallParameterExtractor.java

```diff
@@ -0,0 +1,168 @@
+import org.camunda.feel.syntaxtree.PositionalFunctionParameters;
+import org.camunda.feel.syntaxtree.Ref;
+import scala.jdk.javaapi.CollectionConverters;
+
+class FromAiFeelFunctionCallParameterExtractor implements FeelFunctionCallParameterExtractor {
+
+  @Override
+  public String functionName() {
+    return "fromAi";
+  }
+
+  @Override
+  public AdHocActivityParameter mapToParameter(final FunctionInvocation functionInvocation) {
+    return switch (functionInvocation.params()) {
+      case final PositionalFunctionParameters positionalFunctionParameters ->
+          fromPositionalFunctionInvocationParams(
+              CollectionConverters.asJava(positionalFunctionParameters.params()));
+
+      case final NamedFunctionParameters namedFunctionParameters ->
+          fromNamedFunctionInvocationParams(
+              CollectionConverters.asJava(namedFunctionParameters.params()));
+
+      default ->
+          throw new IllegalArgumentException(
+              "Unsupported fromAi() function invocation: " + functionInvocation.params());
+    };
+  }
+
+  private AdHocActivityParameter fromPositionalFunctionInvocationParams(final List<Exp> params) {
+    final Function<Integer, Exp> getParam =
+        index -> (params.size() > index ? params.get(index) : null);
+
+    return fromFunctionInvocationParams(
+        getParam.apply(0),
+        getParam.apply(1),
+        getParam.apply(2),
+        getParam.apply(3),
+        getParam.apply(4));
+  }
+
+  private AdHocActivityParameter fromNamedFunctionInvocationParams(final Map<String, Exp> params) {
+    return fromFunctionInvocationParams(
+        params.get("value"),
+        params.get("description"),
+        params.get("type"),
+        params.get("schema"),
+        params.get("options"));
+  }
+
+  private AdHocActivityParameter fromFunctionInvocationParams(
+      final Exp name, final Exp description, final Exp type, final Exp schema, final Exp options) {
+
+    final var parameterName = parameterName(name);
+    final var descriptionStr = asString(description, "description");
+    final var typeStr = asString(type, "type");
+    final var schemaMap = asMap(schema, "schema");
+    final var optionsMap = asMap(options, "options");
+
+    return new AdHocActivityParameter(
+        parameterName, descriptionStr, typeStr, schemaMap, optionsMap);
+  }
+
+  private String parameterName(final Exp value) {
+    if (!(value instanceof final Ref valueRef)) {
+      throw new IllegalArgumentException(
+          "Expected fromAi() parameter 'value' to be a reference (e.g. 'toolCall.customParameter'), but received %s."
+              .formatted(
+                  switch (value) {
+                    case final ConstString stringResult ->
+                        "string '%s'".formatted(stringResult.value());
+                    default -> typeMismatchExceptionValue(value);
+                  }));
+    }
+
+    return valueRef.names().last();
```

Comment:
We always take the last segment of a reference into consideration, so for example

`fromAi(toolCall.myValue)`, `fromAi(toolCall.a.b.c.myValue)` and `fromAi(myValue)`

would all be mapped to a parameter name `myValue`, without considering previous segments in the expression.

If we want to reduce ambiguity we could also return the full path (e.g. concatenated by `.`) and it would be in the consumer's responsibility to process the name as needed. WDYT?